docs/wiki/The context - CARML CI environment.md

```diff
@@ -0,0 +1,63 @@
+This page provides an overview of the CARML CI environment. For further details refer to the dedicated Wiki section [The CI environment](./The%20CI%20environment)
+
+---
+
+### _Navigation_
+
+- [Deployment flow](#deployment-flow)
+  - [Module versioning](#module-versioning)
+- [Where does the CARML CI environment fit in?](#where-does-this-platform-fit-in)
+
+---
+
+# Deployment flow
+
+This paragraph provides an overview of the usual deployment flow that goes from source modules to target solutions.
+
+<img src="media\Context\Deployment_flow.png" alt="Deployment flow" height="300">
+
+When it comes to Infrastructure as Code (IaC) deployments, the flow generally covers 3 phases:
+
+1. In the **Develop modules** phase modules are first implemented/updated and then validated using one or multiple test-parameter files, testing their successful deployment against a sandbox subscription to prove their correctness.
+
+1. The next phase, **Publish modules**, publishes the tested and approved modules to a target location for them to be later consumed. The target location should support versioning to allow referencing a specific module version and to avoid breaking changes when referencing them.
+
+1. In the final **Consume modules** phase published modules are referenced and combined to deploy more complex architectures such as services or workloads.
+
+## Module versioning
+
+Deploying resources by referencing their corresponding modules from source control has one major drawback: If your deployments rely on what you have in your source repository then they will by default use the latest code.
+
+Applying software development lifecycle concepts like _publishing build artifacts and versioning_ enables you to have a point in time version of a module. By introducing versions to your modules, the consuming orchestration can and should specify a module version it wants to use and deploy the Azure environment using them.
+
+In case a breaking change is introduced and an updated version is published, no deployments are affected because they still reference the previously published version. Instead, they must make the deliberate decision to upgrade the module to reference newer versions.
```

Comment:
```suggestion
In case a breaking change is introduced and an updated version is published, no deployments are affected because they still reference the previously published version. Instead, they must make the deliberate decision to upgrade the module to reference newer versions.

Also, if you reference a module version that was tested in and has passed through the CI environment , you can trust that it complies the qualitative and functional standards.
```